src/lib.rs

```diff
@@ -255,6 +279,25 @@ pub fn encode_config_buf(input: &[u8], config: Config, buf: &mut String) {
             raw.push(0x3d);
         }
     }
+
+    if config.line_size.is_some() && config.line_size.unwrap() > 0 {
+        let line_size = config.line_size.unwrap();
+        let len = raw.len();
+        let mut i = 0;
+        let mut j = 0;
+
+        while i < len {
+            if i > 0 && i % line_size == 0 {
+                match config.line_ending {
+                    LineEnding::LF => { raw.insert(j, b'\n'); j += 1; }
+                    LineEnding::CRLF => { raw.insert(j, b'\r'); raw.insert(j + 1, b'\n'); j += 2; }
```

Comment:
This would be quadratic in the output length, right? It might not be worth the effort since I can't think of a case where line-wrapped base64 is used for large volumes of data, but a linear-time way of doing this would be to encode smaller chunks of input to get to at least the desired line length worth of encoded text, insert line endings as needed, save the leftovers for the next line, repeat. Every once in a while the leftovers themselves will probably get to be a line of their own, depending on the specific line length. However, I'm not convinced that for the likely use case (decoding a PEM, I imagine) it's really worth it. Also, people who need speed could always hand-roll that around the lower-level logic.